custom/icds/models.py

```diff
@@ -38,9 +38,24 @@ class CCZHosting(models.Model):
     app_id = models.CharField(max_length=255, null=False)
     version = models.IntegerField(null=False)
 
+    def to_json(self, app_names):
+        from custom.icds.serializers import CCZHostingSerializer
+        return CCZHostingSerializer(self, context={'app_names': app_names}).data
+
     def build_doc(self):
         return get_build_by_version(self.link.domain, self.app_id, self.version)
 
+    @classmethod
+    def update_version(cls, link_id, app_id, version):
```

Comment:
this method was later removed and i used simple model save.

But to answer your question, if i remember it correctly it was to call `full_clean` because its not called otherwise i.e related to validations added on the model that need to be checked.